server/models/DbConnect.js

```diff
@@ -0,0 +1,113 @@
+import { Pool } from 'pg';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const createTable = `
+
+
+
+CREATE TABLE IF NOT EXISTS users 
+(user_id SERIAL PRIMARY KEY NOT NULL, 
+    user_name VARCHAR NOT NULL, 
+    user_password VARCHAR NOT NULL,
+    user_role VARCHAR NOT NULL DEFAULT 'user',
+    user_email VARCHAR UNIQUE, 
+    user_address VARCHAR, 
+    user_phone VARCHAR UNIQUE, 
+    user_image VARCHAR);
+
+CREATE TABLE IF NOT EXISTS meals
+    (meal_id SERIAL PRIMARY KEY NOT NULL,
+    meal_price VARCHAR,
+    meal_image VARCHAR,
+    meal_category VARCHAR,
+    meal_added_date TIMESTAMP NOT NULL DEFAULT NOW()
+  ); 
+    
+CREATE TABLE IF NOT EXISTS orders(
+    order_id SERIAL PRIMARY KEY NOT NULL,
+    user_id int REFERENCES users(user_id),
+    meal_id int REFERENCES meals(meal_id),
+    order_phone VARCHAR,
+    order_address VARCHAR,
+    order_data JSON NOT NULL,
+    order_added_date TIMESTAMP NOT NULL DEFAULT NOW(),
+    order_status VARCHAR DEFAULT 'new',
+    order_accepted_time TIMESTAMP DEFAULT NULL,
+    order_decline_time TIMESTAMP DEFAULT NULL,
+    order_completed_time TIMESTAMP DEFAULT NULL
+);`;
+
+let connectionString = '';
+
+if (process.env.NODE_ENV === 'test') {
+  connectionString = process.env.TEST_DB_URL;
+} else {
+  connectionString = process.env.DATABASE_URL || process.env.LOCAL_DB_URL;
+}
+/**
+ * Represents the connection of the app to postgreSql database.
+ */
+class DbConnect {
+  /**
+     * Create database connection.
+     * @param {Promise} pool - takes in the connection string
+     * which is based on the environment
+     */
+  constructor() {
+    this.pool = new Pool({ connectionString });
+  }
+
+  /**
+     * This function creates all needed tables
+     * @returns {Promise} Returns promise that creates all table if resolved
+     * or error if rejected
+     */
+  createAllTables() {
+    this.pool.query(createTable)
+      .then(() => {
+        console.log('Table Created successfully');
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }
+
+  /**
+     * This function connects the app to database base on different environment
+     * @returns {Promise} Returns promise that connect the app to db if resolved
+     * or error if rejected
+     */
+  connectApp() {
+    this.pool.connect()
+      .then(() => {
+        if (process.env.NODE_ENV === 'test') {
+          this.appDelete();
```

Comment:
This is not defined anywhere? Please, can you show me a reference to the definition of this?

Reply:
i renamed the this.appDelete to deleteTables . i must have forgotten to rename it  for the test environment